src/query/parser/interface.go

```diff
@@ -77,5 +79,5 @@ func NewTransformFromOperation(Op Params, nextID int) Node {
 
 // Source represents data sources which are handled differently than other transforms as they are always independent and can always be parallelized
 type Source interface {
-	Execute(ctx context.Context) error
+	Execute(ctx context.Context, queryCtx *models.QueryContext) error
```

Comment:
looks ugly :( Maybe just have QueryContext then

Reply:
I could yeah. Main argument for this kind of signature I think is that `func(ctx context.Context, ...)` is super standard (to the extent that literally any potentially async or remote function probably takes it). That said, I haven't added it to the `Node` subclasses, so maybe better to just go with `queryCtx`.

Reply:
Latest version has things just using `QueryContext`; lmk how it looks.